src/gc.c

```diff
@@ -2739,42 +2739,44 @@ JL_DLLEXPORT void *jl_gc_counted_calloc(size_t nm, size_t sz)
     return b;
 }
 
-JL_DLLEXPORT void jl_gc_counted_free(void *p, size_t sz)
+JL_DLLEXPORT void jl_gc_counted_free_with_size(void *p, size_t sz)
 {
     free(p);
-    gc_num.freed += sz + JL_SMALL_BYTE_ALIGNMENT;
+    gc_num.freed += sz;
     gc_num.freecall++;
 }
 
+JL_DLLEXPORT void jl_gc_counted_free(void *, size_t) __attribute__ ((alias("jl_gc_counted_free_with_size")));
```

Comment:
Dunno and have no way to test. A wrapper function should always work as a fallback though.